### ISSUE TITLE
while inside functions is not coloured correctly

### ISSUE THREAD

**Issue by balupton** [jeff-hykin/better-shell-syntax#83]
## The code with a problem is:

``` bash
#!/usr/bin/env bash

function eval_wrapper {
	while test "$1" = '--'; do
		shift
	done
}
```

``` bash
#!/usr/bin/env bash

function eval_wrapper () (
	while test "$1" = '--'; do
		shift
	done
)

```

## It looks like:
![CleanShot 2024-03-29 at 13 07 51@2x](https://github.com/jeff-hykin/better-shell-syntax/assets/61148/ff6dc6fe-ccf3-4fae-87bb-61b7f945ef5f)
![CleanShot 2024-03-29 at 13 09 47@2x](https://github.com/jeff-hykin/better-shell-syntax/assets/61148/c127ded4-dea9-4d80-958a-96ebf7e751c8)

## It should look like:
![CleanShot 2024-03-29 at 13 08 18@2x](https://github.com/jeff-hykin/better-shell-syntax/assets/61148/6d398075-b7bd-4dbd-bded-b94102e38a38)

## version
v1.8.3

**Comment by Zhengqbbb** [comment]
# same problem - [theme: "Vitesse Theme QB" ]

**It seems that the first line of "while" in the function will cause**

```sh
testing(){
	while :; do
		echo 1
	done
}
```

![CleanShot 2024-03-29 at 16 37 31@2x](https://github.com/jeff-hykin/better-shell-syntax/assets/40693636/f82c953c-d855-4b35-b146-3e23818fbd7d)

**Comment by jeff-hykin** [comment]
should be fixed with v1.8.5
